modules/api/src/main/java/org/apache/fluo/api/client/LoaderExecutor.java

```diff
@@ -48,11 +48,12 @@
 
   /**
    * Same as {@link #execute(Loader)} except it returns a future that completes upon successful
-   * commit and if an exception is thrown in the loader, it will be relayed through the future.
+   * commit and if an exception is thrown in the loader, it will be relayed through the future. The
+   * result of the future is the Loader that was successfully executed.
    *
    * @since 2.0.0
    */
-  CompletableFuture<Void> submit(Loader loader);
+  CompletableFuture<Loader> submit(Loader loader);
```

Comment:
I am not sure of the exact syntax or if its possible, but if this could be tweaked allow user to use their own types it could avoid the user having to cast.  Something like (syntax is wrong):

```java
T extends Loader CompletableFuture<T> submit(T loader)
```

I am going to circle back and look into proper syntax and semantics later today.

Reply:
I think the syntax would be:

```suggestion
  <T extends Loader> CompletableFuture<T> submit(T loader);
```

Or possibly:

```suggestion
  <T extends Loader> CompletableFuture<? super T> submit(T loader);
```

Reply:
Looks like your suggestion works great. Much nicer without the casting. Thanks for the improvement! I will get a new commit in later today

Reply:
@ctubbsii  I tried both of your suggestions locally w/ this branch.   The first one `<T extends Loader> CompletableFuture<T> submit(T loader);` seems to work while the other one did not.  

I used the IT added by this PR to experiment and the following compiles with the first suggestion and does not require the casts.

```java
    CompletableFuture<CounterLoader> future = le.submit(new CounterLoader());
    Assert.assertEquals(future.get().getNewValue().longValue(), 1);

    future = le.submit(new CounterLoader());
    Assert.assertEquals(future.get().getNewValue().longValue(), 2);
    
    // ensure passing Loader type itself and not a subtype of Loader works
    Loader someLoader = new CounterLoader();
    CompletableFuture<Loader> future2 = le.submit(someLoader);
    Assert.assertEquals(((CounterLoader)future2.get()).getNewValue().longValue(), 3);
```

When trying the 2nd suggestion w/ the above code, I see errors like the following.

```
Type mismatch: cannot convert from CompletableFuture<capture#2-of ? super LoaderExecutorIT.CounterLoader> to CompletableFuture<LoaderExecutorIT.CounterLoader>	LoaderExecutorIT.java	/fluo-integration-tests/src/main/java/org/apache/fluo/integration/client	line 149	Java Problem
```

Reply:
Hmm, it might be possible to do:

```suggestion
  <T extends Loader,S super T> CompletableFuture<S> submit(T loader);
```

It might not be worth it to use `super` at all, though.

Reply:
The following does not compile 

```java
<T extends Loader,S super T> CompletableFuture<S> submit(T loader);
```

@ctubbsii  I tried the following w/ your first suggestion and it worked great.

```
 CompletableFuture<Loader> future = le.submit(new CounterLoader());
```

The first suggestions it looking really good. Is there something you were trying to accomplish with the other suggestions?

Reply:
I was thinking about how you can't do:

```java
  ArrayList<Number> numbers = new ArrayList<Integer>();
```

I always have to do a deep dive every time I refresh my memory on generics and the use of `super`. I commented above without that refresher and without trying the suggestions myself. Please use whatever works and disregard the other suggestions.